### PR TITLE
Add logging & more tests

### PR DESCRIPTION
**Pull request by knapii-developments**
Adding more/better logging statements
Adding tests:
* When the key is already disabled, Yale should not try to disable the key again
* Yale should not delete key before it's time to

### CODE REVIEW

File: internal/yale/delete.go [hunk 1]
```diff
@@ -41,26 +41,32 @@ func (m *Yale) DeleteKey(k8Secret *corev1.Secret, gcpSaKeySpec apiv1b1.GCPSaKeyS
...
 	saName := secretAnnotations["serviceAccountKeyName"]
 	saKey, err := m.GetSAKey(saName, keyName)
+	keyNameForLogs := after(saKey.serviceAccountKeyName, "serviceAccounts/")
 	if err != nil {
 		return err
 	}
+	logs.Info.Printf("Checking if %s should be deleted.", keyNameForLogs)
 	totalTime := gcpSaKeySpec.KeyRotation.DisableAfter + gcpSaKeySpec.KeyRotation.DeleteAfter
-	isInUse, err := m.IsAuthenticated(totalTime, keyName, gcpSaKeySpec.GoogleServiceAccount.Project)
+	isNotUsed, err := m.IsAuthenticated(totalTime, keyName, gcpSaKeySpec.GoogleServiceAccount.Project)
```
Comment (jyang-broad) [inline internal/yale/delete.go:50]: Naming nit: without looking at the code. I'd expect something named IsAuthenticated to return a boolean. true -> authenticated, false -> unathenticated. I'm kind of confused why you'd name the result to be "isNotUsed".

File: internal/yale/disable.go [hunk 2]
```diff
@@ -97,23 +124,26 @@ func (m *Yale) IsAuthenticated(timeSinceAuth int, keyName string, googleProject
...
+		logs.Info.Printf("%s is being used.", keyNameForLogs)
+	}
+	return keyIsNotUsed, err
 }
 
 // IsExpired Determines if it's time to disable a key
-func IsExpired(beginDate string, duration int, keyName string) (bool, error) {
-	dateAuthorized, err := time.Parse(time.RFC3339, beginDate)
+func IsExpired(beginDate string, duration int) (bool, error) {
+	formatedBeginDate, err := time.Parse(time.RFC3339, beginDate)
```
Comment (jyang-broad) [inline internal/yale/disable.go:138]: sp: `formattedBeginDate`

File: internal/yale/rotate.go [hunk 3]
```diff
@@ -23,6 +24,10 @@ const KEY_ALGORITHM string = "KEY_ALG_RSA_2048"
 // KEY_FORMAT format to use when creating new Google SA keys
 const KEY_FORMAT string = "TYPE_GOOGLE_CREDENTIALS_FILE"
 
+// Returns service account name
+// Ex: agora-perf-service-account@broad-dsde-perf.iam.gserviceaccount.com returns agora-perf-service-account
```
Comment (jyang-broad) [inline internal/yale/rotate.go:28]: nice comment, love the real-world example

**Review by jyang-broad — APPROVED** [review]
Looks Sane! Left some sp/nits.

File: internal/yale/disable.go (hunk 2 above)
Comment (knapii-developments) [reply internal/yale/disable.go]: 30adc2a

File: internal/yale/delete.go (hunk 1 above)
Comment (knapii-developments) [reply internal/yale/delete.go]: 30adc2a

**Pull request merged**